### PR TITLE
Synchronous Hyperband

### PR DESCRIPTION
**Pull request by mseeger**
*Issue #, if available:*
Inherited from #18 

*Description of changes:*
Adding some tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mseeger** [comment]
I'll still add one more test, otherwise this is good to go. Had to open another PR due to renaming.

**Comment by mseeger** [comment]
Hello, I now implemented the final test (I extended test_scheduler_api.py for sync schedulers). Along the way, I also made sure on_trial_error is correctly implemented.

File: tst/schedulers/test_schedulers_api.py [hunk 1]
```diff
@@ -96,3 +99,72 @@ def test_schedulers_api(scheduler):
...
+    SynchronousGeometricHyperbandScheduler(
+        config_space=config_space,
+        max_resource_level=max_t,
+        brackets=3,
+        resource_attr=resource_attr,
+        batch_size=sync_batch_size,
+        metric=metric1,
+        max_resource_attr='steps'),
+])
+def test_sync_schedulers_api(scheduler):
```
Comment (geoalgo) [inline tst/schedulers/test_schedulers_api.py:113]: This solution works for me, is there a particular reason why we would not test this scheduler among others? It should still obey the same API right?

**Review by geoalgo — APPROVED** [review]

File: tst/schedulers/test_schedulers_api.py (hunk 1 above)
Comment (mseeger) [reply tst/schedulers/test_schedulers_api.py:113]: Good question: A sync scheduler with a certain batch_size (say: 4) can expect to receive batch_size suggest calls before any on_trial_result calls. You can expect there is a suggest and a collect phase. At least, this is what the Tuner code implies. It really calls suggest batch_size ( = num_workers) time in a row, and only later then queries results.

This is how I implemented sync HB, it is simple and clean that way. You are right that *some* sync schedulers do not make use of this (e.g., FIFOScheduler can be run sync).

**Pull request merged**